extensions/sql-bindings/src/services/azureFunctionsService.ts

```diff
@@ -46,37 +46,54 @@ export async function createAzureFunction(node?: ITreeNodeInfo): Promise<void> {
 				.withAdditionalProperties(propertyBag).send();
 
 			// prompt user for connection profile to get connection info
-			quickPickStep = 'getConnectionInfo';
-			connectionInfo = await vscodeMssqlApi.promptForConnection(true);
-			if (!connectionInfo) {
-				// User cancelled
-				return;
-			}
-			TelemetryReporter.createActionEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.startCreateAzureFunctionWithSqlBinding)
-				.withAdditionalProperties(propertyBag).withConnectionInfo(connectionInfo).send();
-
-			// list databases based on connection profile selected
-			connectionURI = await vscodeMssqlApi.connect(connectionInfo);
-			listDatabases = await vscodeMssqlApi.listDatabases(connectionURI);
-			const selectedDatabase = (await vscode.window.showQuickPick(listDatabases, {
-				canPickMany: false,
-				title: constants.selectDatabase,
-				ignoreFocusOut: true
-			}));
+			while (true) {
+				connectionInfo = await vscodeMssqlApi.promptForConnection(true);
+				if (!connectionInfo) {
+					// User cancelled
+					return;
+				}
+				quickPickStep = 'getConnectionInfo';
+				try {
+					await vscode.window.withProgress(
+						{
+							location: vscode.ProgressLocation.Notification,
+							title: constants.connectionProgressTitle,
+							cancellable: false
+						}, async (_progress, _token) => {
+							// list databases based on connection profile selected
+							if (!connectionInfo) {
+								// User cancelled
+								return;
+							}
+							connectionURI = await vscodeMssqlApi.connect(connectionInfo);
+						}
+					);
+				} catch (e) {
+					// User cancelled
+					continue;
+				}
+				// list databases based on connection profile selected
+				listDatabases = await vscodeMssqlApi.listDatabases(connectionURI);
```

Comment:
What about the offline scenario? We still want to allow users to manually enter in the DB if they want so we should add an option for them to do so.

Reply:
We can track the offline scenario in a separate follow-up issue. But that is currently not the focus. We plan to add more with the table prompts so I would see this coming in when we address that.